src/SFA.DAS.AssessorService.Application/Handlers/OrganisationStandards/OrganisationStandardVersionOptInHandler.cs

```diff
@@ -0,0 +1,108 @@
+            _mediator = mediator;
+            _unitOfWork = unitOfWork;
+            _logger = logger;
+        }
+
+        public async Task<OrganisationStandardVersion> Handle(OrganisationStandardVersionOptInRequest request, CancellationToken cancellationToken)
+        {
+            try
+            {
+                _unitOfWork.Begin();
+
+                OrganisationStandardVersion orgStandardVersion = null;
+
+                var orgStandard = await _repository.GetOrganisationStandardByOrganisationIdAndStandardReference(request.EndPointAssessorOrganisationId, request.StandardReference);
+                if (orgStandard != null)
+                {
+                    var entity = new Domain.Entities.OrganisationStandardVersion
+                    {
+                        StandardUId = request.StandardUId,
+                        Version = request.Version,
+                        OrganisationStandardId = orgStandard.Id,
+                        EffectiveFrom = request.EffectiveFrom,
+                        EffectiveTo = request.EffectiveTo,
+                        DateVersionApproved = request.DateVersionApproved,
+                        Comments = request.Comments,
+                        Status = request.Status
+                    };
+
+                    await _repository.CreateOrganisationStandardVersion(entity);
+                    orgStandardVersion = (OrganisationStandardVersion)entity;
+
+                    var application = await _applyRepository.GetApply(request.ApplicationId);
+                    var submittingContact = await _contactQueryRepository.GetContactById(request.SubmittingContactId);
+                    var standard = await _standardRepository.GetStandardVersionByStandardUId(request.StandardUId);
+
+                    application.ApplicationStatus = ApplicationStatus.Approved;
+                    application.ReviewStatus = ApplicationStatus.Approved;
+                    application.StandardReference = request.StandardReference;
+                    application.StandardCode = standard.LarsCode;
+                    application.ApplyData.Apply.StandardCode = standard.LarsCode;
+                    application.ApplyData.Apply.StandardReference = request.StandardReference;
+                    application.ApplyData.Apply.StandardName = standard.Title;
+                    application.ApplyData.Apply.Versions = new List<String>() { request.Version.ToString() };
+
+                    await _applyRepository.SubmitApplicationSequence(application);
+
+                    await NotifyContact(submittingContact, application.ApplyData, cancellationToken);
+
+                    _unitOfWork.Commit();
+                }
+                    
+                return orgStandardVersion;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Failed to opt-in standard version {request.StandardReference} for Organisation {request.EndPointAssessorOrganisationId}");
```

Comment:
do we want to log which version they were trying to opt in for?

Reply:
Fixed